core/eolearn/core/eodata.py

```diff
@@ -699,7 +699,7 @@ def consolidate_timestamps(self, timestamps: List[dt.datetime]) -> Set[dt.dateti
         good_timestamp_idxs = [idx for idx, _ in enumerate(self.timestamps) if idx not in remove_from_patch_idxs]
         good_timestamps = [date for idx, date in enumerate(self.timestamps) if idx not in remove_from_patch_idxs]
 
-        relevant_features = set(FeatureTypeSet.TEMPORAL_TYPES) - {FeatureType.TIMESTAMPS}
+        relevant_features = filter(lambda ftype: ftype.is_temporal() and ftype != FeatureType.TIMESTAMPS, FeatureType)
```

Comment:
might be more elegant:

```suggestion
        relevant_features = filter(lambda ftype: ftype.is_temporal() and not ftype.is_meta(), FeatureType)
```